src/vmspawn/vmspawn.c

```diff
@@ -186,6 +192,20 @@ static int help(void) {
         return 0;
 }
 
+static int parse_environment(void) {
```

Comment:
Or, rename this to more specific, e.g. parse_mac_address_from_env(), and call before generating MAC address.